buf/registry/plugin/v1beta1/collection_service.proto

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.registry.plugin.v1beta1;
+
+import "buf/registry/plugin/v1beta1/collection.proto";
+import "buf/registry/plugin/v1beta1/plugin.proto";
+import "buf/validate/validate.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/plugin/v1beta1";
+
+// Operate on Collections.
+service CollectionService {
+  // Get Collections.
+  rpc GetCollections(GetCollectionsRequest) returns (GetCollectionsResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // List Collections for a given Plugin.
+  rpc ListCollections(ListCollectionsRequest) returns (ListCollectionsResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
+  // Get the Collections for the given Plugins.
+  rpc GetCollectionsPlugins(GetCollectionsPluginsRequest) returns (GetCollectionsPluginsResponse) {
```

Comment:
@bufdev not sure how much you hate this, but as @emcfarlane and I flushed this model out it seemed better to remove `collectios_ids` from the Plugin entity and instead have a flattened list via this RPC.

Happy to bring back `collection_ids` on Plugin, but figured I'd add this for discussion. Say if you have PluginRefs and want to know which collection each of those belongs to you would:

1. Call this endpoint with plugin refs
2. Resolve the set of `collection_id` against `GetCollections`